src/Common/src/Interop/Unix/System.Native/Interop.RealPath.cs

```diff
@@ -0,0 +1,44 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.InteropServices;
+
+internal static partial class Interop
+{
+    internal static partial class Sys
+    {
+        /// <summary>
+        /// Takes a path containing relative subpaths or links and returns the absolute path.
+        /// </summary>
+        /// <param name="path">The input path containing relative subpaths or links</param>
+        /// <param name="resultPath">The output path; will be null on error</param>
+        /// <returns>Returns 0 on succes and -1 on failure; on failure, resultPath will be null</returns>
+        internal static int RealPath(string path, out string resultPath)
+        {
+            byte[] buffer = new byte[Interop.Sys.MaxPath];  // MaxPath should be big enough every time
+            int result = RealPath(path, buffer, buffer.Length);
+            if (result < 0)
+            {
+                resultPath = null; 
+            }
+            else
+            {
+                resultPath = System.Text.Encoding.UTF8.GetString(buffer, 0, buffer.Length);
```

Comment:
I don't think this is right.  Passing in buffer.Length is going to cause it to walk the whole buffer and include all of that data in the string.  i.e. this:
```C#
byte[] data = new byte[100];
string s = Encoding.UTF8.GetString(data, 0, data.Length);
Console.WriteLine(s.Length);
```
prints out 100, not 0.